### PR TITLE
Add float16 support to dropout operator

### LINKED ISSUE #9222: Need float16 support in dropout operator

### PR DESCRIPTION
**Pull request by kexinzhao**
fix #9222 

Added device function for multiplying two float16 numbers on GPU device, which is needed in the following code:
``` cpp
Y.device(place) = X * static_cast<T>(1.0f - dropout_prob);
```

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_dropout_op.py [hunk 1]
```diff
@@ -82,5 +83,37 @@ def test_check_output(self):
...
+        out = x * (1.0 - prob)
+
+        self.op_type = "dropout"
+        self.inputs = {'X': OpTest.np_dtype_to_fluid_dtype(x)}
+        self.attrs = {'dropout_prob': prob, 'is_test': True}
+        self.outputs = {'Out': out}
+
+    def test_check_output(self):
+        if core.is_compiled_with_cuda() and core.op_support_gpu("dropout"):
+            self.check_output_with_place(core.CUDAPlace(0), atol=1e-3)
```
Comment (chengduoZH) [inline python/paddle/fluid/tests/unittests/test_dropout_op.py]: `TestFP16DropoutOp1` and `TestFP16DropoutOp2` are very similar, and they can be inherited relationships, which can reduce the amount of code.

Comment (kexinzhao) [reply python/paddle/fluid/tests/unittests/test_dropout_op.py]: You are right! Done.

File: paddle/fluid/platform/float16.h [hunk 2]
```diff
@@ -668,71 +783,71 @@ HOST inline bool operator>=(const float16& a, const float16& b) {
   return (res & 0xffff) != 0;
 }
 
-// Arithmetic operators, software emulated on other CPU
+// Arithmetic operators for float16, software emulated on other CPU
 #else
-HOSTDEVICE inline float16 operator+(const float16& a, const float16& b) {
+HOST inline float16 operator+(const float16& a, const float16& b) {
```
Comment (chengduoZH) [inline paddle/fluid/platform/float16.h]: Maybe `HOST` is unnecessary.

Comment (kexinzhao) [reply paddle/fluid/platform/float16.h]: I guess so. Let me fix that in the next PR.

**Pull request merged**